### PR TITLE
Bring doc improvements for v3 across from master

### PR DESCRIPTION
**Pull request by mikehearn**
This PR includes docs for contract constraints (Tudor/Michele's work), some more on contract upgrades, some network map doc improvements and some more discussion of flow drains.

### CODE REVIEW

**Comment by mikehearn** [comment]
These PRs were already reviewed by Shams and Joel. This PR just rebases them.

File: docs/source/api-contract-constraints.rst [hunk 1]
```diff
@@ -127,6 +195,10 @@ to use as CorDapps using the ``cordappPackages`` parameter.
 
     MockServices mockServices = new MockServices(Arrays.asList("com.domain.cordapp"))
 
+However - there is an easier way! If your unit tests are in the same package as the contract code itself, then you
```
Comment (fenryka) [inline docs/source/api-contract-constraints.rst]: However<comma>, unless the However<dash> is idimatic

File: docs/source/network-map.rst [hunk 2]
```diff
@@ -92,16 +124,23 @@ Network parameters update process
 ---------------------------------
 
 In case of the need to change network parameters Corda zone operator will start the update process. There are many reasons
-that may lead to this decision: we discovered that some new fields have to be added to enable smooth network interoperability or change
-of the existing compatibility constants is required due to upgrade or security reasons.
+that may lead to this decision: adding a notary, setting new fields that were added to enable smooth network interoperability,
+or a change of the existing compatibility constants is required, for example.
+
+.. note:: A future release may support the notion of phased rollout of network parameter changes.
```
Comment (fenryka) [inline docs/source/network-map.rst]: of *a* phased rollout

File: docs/source/network-map.rst [hunk 3]
```diff
@@ -110,12 +149,22 @@ node operator about the event.
         :start-after: DOCSTART 1
         :end-before: DOCEND 1
 
-Node administrator can review the change and decide if is going to accept it. The approval should be done before ``updateDeadline``.
-Nodes that don't approve before the deadline will be removed from the network map.
-If the network operator starts advertising a different set of new parameters then that new set overrides the previous set. Only the latest update can be accepted.
-To send back parameters approval to the zone operator RPC method ``fun acceptNewNetworkParameters(parametersHash: SecureHash)``
-has to be called with ``parametersHash`` from update. Notice that the process cannot be undone.
+The node administrator can review the change and decide if they are going to accept it. The approval should be do
```
Comment (fenryka) [inline docs/source/network-map.rst]: done, not do

File: docs/source/network-map.rst [hunk 4]
```diff
@@ -110,12 +149,22 @@ node operator about the event.
...
+before the update Deadline. Nodes that don't approve before the deadline will likely be removed from the network map by
```
Comment (fenryka) [inline docs/source/network-map.rst]: deadline not Deadline?

File: docs/source/network-map.rst [hunk 5]
```diff
@@ -110,12 +149,22 @@ node operator about the event.
...
+the zone operator, but that is a decision that is left to the operator's discretion. For example the operator might also
```
Comment (fenryka) [inline docs/source/network-map.rst]: For example <comma>

I *think*

**Pull request merged**